test/http_fakes.dart

```diff
@@ -29,6 +29,7 @@ class FakeHttpHeaders implements HttpHeaders {
 
   @override
   set ifModifiedSince(DateTime? ifModifiedSince) {
+    ArgumentError.checkNotNull(ifModifiedSince);
```

Comment:
Since we have this check now I would modify the section below to use `!`, without the `if`.

Reply:
Done.